### PR TITLE
Update deprecated and broken links

### PR DESCRIPTION
**Pull request by benmoss**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Fixes a whole bunch of broken links across the codebase.

**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:
It seems that some of the KEPs have been moved to reflect the new numbering scheme, but others are still there under the old one. I figured it was good enough to fix these links now and then we can circle back if they break again.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/update.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ if ! ${ALL} ; then
 	echo "Running in short-circuit mode; run with FORCE_ALL=true to force all scripts to run."
 fi
 
-"${KUBE_ROOT}/hack/godep-restore.sh" ${V}
+# "${KUBE_ROOT}/hack/godep-restore.sh" ${V}
```
Comment (BenTheElder) [inline hack/make-rules/update.sh]: seems unrelated to the stated purpose?

**Comment by rlenferink** [comment]
I'm not an approver on k8s/k8s but the changes look good to me

/lgtm

**Comment by benmoss** [comment]
/assign @erictune 
/assign @ktsakalozos
/assign @Random-Liu
/assign @dims 
/assign @smarterclayton 

apologies for the assignment blast, but it seems to be the only way to get this moved along.

**Comment by tallclair** [comment]
Thanks. Should we build some automation for this kind of thing? It's only a matter of time until these are out of date again...
/approve

**Comment by benmoss** [comment]
> Thanks. Should we build some automation for this kind of thing? It's only a matter of time until these are out of date again...

Yeah, I was thinking about this. I had a shell pipeline I tried to use for finding these, but I'm sure it could be automated and improved.

Finding 404s:
```
for match in $(ag --nogroup --nonumbers --noheading -o "(https:\/\/github\.com\/kubernetes\/[\w\/\-\.]*)" | grep -Ev "issues|pull|generated" | cut -d":" -f2- | sort | uniq); do if ! curl -L -sfo /dev/null $match; then; echo "got a match $match"; fi; done
```

Finding "this file has moved":
```
for match in $(ag --nogroup --nonumbers --noheading -o "(https:\/\/github\.com\/kubernetes\/[\w\/\-\.]*)" | grep -Ev "issues|pull|generated" | cut -d":" -f2- | sort | uniq); do echo $match; if curl -L -sfo /dev/null $match | grep -q "This file has moved"; then; echo "got a match $match"; fi; done
```

File: cluster/addons/dns/nodelocaldns/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 This addon runs a node-local-dns pod on all cluster nodes. The pod runs CoreDNS as the dns cache. It runs with `hostNetwork:True` and creates a dedicated dummy interface with a link local ip(169.254.20.10/32 by default) to listen for DNS queries. The cache instances connect to clusterDNS in case of cache misses.
 
-Design details [here](https://github.com/kubernetes/community/blob/master/keps/sig-network/0030-nodelocal-dns-cache.md)
+Design details [here](https://git.k8s.io/enhancements/keps/sig-network/0030-nodelocal-dns-cache.md)
```
[the remark this answers is not in the document]
Comment (benmoss) [reply cluster/addons/dns/nodelocaldns/README.md]: I don't think I understand. Are you saying that the old link redirects to that, or that the new one does? I agree that the new one does, that's what it's supposed to do. Is that a problem?

[the remark this answers is not in the document]
Comment (mourya007) [reply cluster/addons/dns/nodelocaldns/README.md]: No, I think I got it

**Comment by nikhita** [comment]
@benmoss PR needs rebase.

**Comment by smarterclayton** [comment]
Looks ok, the api changes look fine.

/approve
